lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -515,12 +556,15 @@ def authorize_ip(type, changed, client, group, groupRules,
     return changed, ip_permission
 
 
-def serialize_group_grant(group_id, rule):
+def serialize_group_grant(group_id, rule, has_rule_description=None):
     permission = {'IpProtocol': rule['proto'],
                   'FromPort': rule['from_port'],
                   'ToPort': rule['to_port'],
                   'UserIdGroupPairs': [{'GroupId': group_id}]}
 
+    if has_rule_description:
```

Comment:
you don't need `has_rule_description` for this method (or `serialize_ip_grant`).

There are two cases:
* There are no rule descriptions
* There are rule descriptions and botocore supports that (otherwise `fail_json` was called before we got here)

On that basis, you can do this here instead:

```
if 'rule_desc' in rule:
    permission['UserIdGroupPairs'][0]['Description'] = rule['rule_desc']
```